### PR TITLE
prefetch weights while waiting for pending requests to complete

### PR DESCRIPTION
**Pull request by JenniferWang**
Summary:
Feature parity with v0: allow prefetching weights while waiting for the pending requests to finish.

## Test Plan 
Introduced a benchmark that simulates the on-going requests with actual weight sync logic.

Reference Group (V0)
```
================================================================================
WEIGHT SYNC BENCHMARK RESULTS
================================================================================
Model: Qwen/Qwen3-8B
Model size: 15.26 GB
Iterations: 3
Prefetch enabled: False
--------------------------------------------------------------------------------
Metric                         Time (s)        Throughput (GB/s)   
--------------------------------------------------------------------------------
Avg push_weights                      5.102 s         2.99 GB/s
Avg update_weights                   43.738 s         0.35 GB/s
Avg total (push + update)            48.840 s
================================================================================

================================================================================
WEIGHT SYNC BENCHMARK RESULTS
================================================================================
Model: Qwen/Qwen3-8B
Model size: 15.26 GB
Iterations: 3
Prefetch enabled: True
Fetcher procs: 8
--------------------------------------------------------------------------------
Metric                         Time (s)        Throughput (GB/s)   
--------------------------------------------------------------------------------
Avg push_weights                      5.208 s         2.93 GB/s
Avg update_weights                   29.602 s         0.52 GB/s
Avg total (push + update)            34.810 s
================================================================================

```

Test Group (V1)
```
================================================================================
WEIGHT SYNC BENCHMARK RESULTS
================================================================================
Model: Qwen/Qwen3-8B
Model size: 15.26 GB
Iterations: 3
Prefetch enabled: False
--------------------------------------------------------------------------------
Metric                         Time (s)        Throughput (GB/s)   
--------------------------------------------------------------------------------
Avg push_weights                      5.070 s         3.01 GB/s
Avg update_weights                   39.974 s         0.38 GB/s
Avg total (push + update)            45.044 s
================================================================================

================================================================================
WEIGHT SYNC BENCHMARK RESULTS
================================================================================
Model: Qwen/Qwen3-8B
Model size: 15.26 GB
Iterations: 3
Prefetch enabled: True
Fetcher procs: 8
--------------------------------------------------------------------------------
Metric                         Time (s)        Throughput (GB/s)   
--------------------------------------------------------------------------------
Avg push_weights                      5.055 s         3.02 GB/s
Avg update_weights                   28.730 s         0.53 GB/s
Avg total (push + update)            33.784 s
================================================================================
```

## Next Steps
[-] implement the prefetch logic & shared memory
[-] Add metric similar to generator v0
[ ] Perf/Throughput testing compared to generator v0

Differential Revision: D91092833

### CODE REVIEW

File: benchmarks/generator/weight_sync.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class BenchmarkTitanTrainer(TitanTrainer):
+    """TitanTrainer with weight modification capabilities for benchmarking."""
+
+    @endpoint
+    async def modify_weights(self, scale: float = 1.1):
```
Comment (joecummings) [inline benchmarks/generator/weight_sync.py]: super nit: do we need to parameterize this? Can't we just 1) assume it's a floating point and 2) arbitrarily add or scale by X ?

File: src/forge/actors/vllm/v1/forge_executor.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    def _init_executor(self) -> None:
+        """Initialize executor and deserialize TorchStore Controller."""
+        super()._init_executor()
+
+        controller_str = os.environ.get("VLLM_TORCHSTORE_CONTROLLER")
+        if controller_str:
+            logger.info(
+                "[ForgeMonarchExecutor] Deserializing TorchStore Controller from environment..."
+            )
+            self.torchstore_controller = cloudpickle.loads(
```
Comment (joecummings) [inline src/forge/actors/vllm/v1/forge_executor.py:219]: 😅

File: benchmarks/generator/weight_sync.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+
+
+@dataclass
+class BenchmarkResults:
+    """Aggregated benchmark results."""
+
+    model: str
+    iterations: int
+    prefetch_enabled: bool
+    n_fetcher_procs: int
```
Comment (joecummings) [inline benchmarks/generator/weight_sync.py:89]: Can we test how this parameter affects throughput?

File: benchmarks/generator/weight_sync.py (hunk 1 above)
Comment (JenniferWang) [reply benchmarks/generator/weight_sync.py]: I'll simplify this. The rest of the test is quite essential.

File: benchmarks/generator/weight_sync.py (hunk 3 above)
Comment (JenniferWang) [reply benchmarks/generator/weight_sync.py:89]: `n_fetcher_procs` -- too high will slightly degrade the overall time. 
`prefetch_enabled` -- this is the major toggle.

Comment (joecummings) [reply benchmarks/generator/weight_sync.py:89]: Right but is there a graph of where "too high" is? I imagine that also too low will not be optimized.
